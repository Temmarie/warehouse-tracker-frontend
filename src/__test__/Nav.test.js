import React from 'react';
import {
  BrowserRouter as Router,
  Link,
} from 'react-router-dom';
import renderer from 'react-test-renderer';

test('Link changes the class when clicked', () => {
  const component = renderer.create(
    <Router>
      <Link to="/Login">Login</Link>
      ,
      <Link to="/Signup">Signup</Link>
      ,
      <Link to="/">LOGOUT</Link>
      ,
    </Router>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
