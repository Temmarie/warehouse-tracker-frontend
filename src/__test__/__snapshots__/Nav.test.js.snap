// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when clicked 1`] = `
Array [
  <a
    href="/Login"
    onClick={[Function]}
  >
    Login
  </a>,
  ",",
  <a
    href="/Signup"
    onClick={[Function]}
  >
    Signup
  </a>,
  ",",
  <a
    href="/"
    onClick={[Function]}
  >
    LOGOUT
  </a>,
  ",",
]
`;

exports[`Link changes the class when clicked 2`] = `
Array [
  <a
    href="/Login"
    onClick={[Function]}
  >
    Login
  </a>,
  ",",
  <a
    href="/Signup"
    onClick={[Function]}
  >
    Signup
  </a>,
  ",",
  <a
    href="/"
    onClick={[Function]}
  >
    LOGOUT
  </a>,
  ",",
]
`;

exports[`Link changes the class when hovered 1`] = `
Array [
  <a
    href="/Login"
    onClick={[Function]}
  >
    Login
  </a>,
  ",",
  <a
    href="/Signup"
    onClick={[Function]}
  >
    Signup
  </a>,
  ",",
  <a
    href="/"
    onClick={[Function]}
  >
    LOGOUT
  </a>,
  ",",
]
`;

exports[`Link changes the class when hovered 2`] = `
Array [
  <a
    href="/Login"
    onClick={[Function]}
  >
    Login
  </a>,
  ",",
  <a
    href="/Signup"
    onClick={[Function]}
  >
    Signup
  </a>,
  ",",
  <a
    href="/"
    onClick={[Function]}
  >
    LOGOUT
  </a>,
  ",",
]
`;
